docs/design/proposals/codegen-refactor/centralize_1.md

```diff
@@ -0,0 +1,197 @@
+* `shape` | `aws-sdk` | `sdk-shape` | `sdk`: the original operations, models, errors, structs for a given AWS service. sourced from *aws-sdk*, ex: [aws-sdk-go s3](https://github.com/aws/aws-sdk-go/blob/main/models/apis/s3/2006-03-01/api-2.json#L1)
+* `reconciliation`: logic involving access to or relations between `resource`, `shape`, `ackgenconfig`, and `aws-sdk`
+* `ackmodel`: the [code](https://github.com/aws-controllers-k8s/code-generator/blob/main/pkg/model/model.go#L36) representation of ACK's view of the world; the source of truth for `aws-sdk`, `ackgenconfig`, and `reconciliation`
+
+## Problem
+It is becoming increasingly difficult to contribute to and maintain *code-generator* due to unclear encapsulations and dispersed logic throughout the codebase. For example, `ackgenconfig` is passed throughout code and accessed in a variety of ways, and logic involving `reconciliation` is inconsistent, both in implementation and interface. Some specific examples include:
+  * [overextended encapsulations](#overextended-encapsulations)
+  * [long function parameter lists](#long-parameters)
+  * [inconsistent use of `ackgenconfig`](#ackgenconfig-use)
+  * [increased cognitive load for contributors](#more-work-for-contributors)
+
+
+## Solution
+The solution is to repair and realign encapsulations in `code-generator/pkg/`:
+  * enforce `ackgenconfig` is **only** accessible via *Getters* exposed in [config package](https://github.com/aws-controllers-k8s/code-generator/tree/25c43e827527b43c652b6e1995265c8d6027567f/pkg/generate/config).  
+  * centralize `reconciliation` by extending `ackmodel` in [model.go](https://github.com/aws-controllers-k8s/code-generator/blob/25c43e827527b43c652b6e1995265c8d6027567f/pkg/model/model.go#L36) 
+  * update accessibility of *Setters* so only the relevant interfaces are public
+    * ex: [SetResourceForStruct](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/pkg/generate/code/set_resource.go#L1217) vs. [setResourceForContainer](https://github.com/aws-controllers-k8s/code-generator/blob/26e5da2e7656bb836ee438c05df14f2adc50197d/pkg/generate/code/set_resource.go#L1154)
+
+
+### Requirements
+* No impact to existing functionality
+* Code is easy to read, intuitive, and extendable
+* Naming in code is consistent (ex: Find vs. Get)
+* Navigating code is easier because pkg responsibility is clear
+
+
+### Prerequisites
+*code-generator* contains unit tests and also runs a subset of $SERVICE-controller e2e tests against PRs. Are these comprehensive enough to detect breakage in the code that we'll be touching/refactoring? If there are gaps between tests and code being refactored, then **tests will need to be created or updated** prior to implementing the solution.
```

Comment:
> Are these comprehensive enough to detect breakage in the code that we'll be touching/refactoring?

We need to identify the possibly affected code paths and determine whether these are being covered in any of our tests